### DISCUSSION TITLE
RobustnessVerificationTreeModelsCliqueMethod & Decision Trees

### DISCUSSION THREAD

**Discussion by ghost** [Trusted-AI/adversarial-robustness-toolbox#1365] [Q&A] [answered]
Hello,
Does the RobustnessVerificationTreeModelsCliqueMethod work for Decision Trees? If not why?
I am getting an error saying 'ScikitlearnDecisionTreeClassifier' object has no attribute 'get_trees'.
Why does the method require an ensemble of trees and does not work for single trees?
Thank you so much,
Melike

**Comment by beat-buesser** [comment, marked as answer]
Hi @melikedem I think it's because the algorithm implemented by `RobustnessVerificationTreeModelsCliqueMethod` is designed for ensembles of trees, for example where it forms cliques. A single tree can be verified exactly and much more efficiently in linear time with a different algorithm (Algorithm 3). Would you be interested to implemented the Algorithm 3 for a single decision tree?

**Reply by ghost** [reply]
Hi @beat-buesser,
I am sorry because of my late answer. Thank you for the explanation, could you please provide a reference for me to learn more about the Algorithm 3? Then I can try my best to implement it.
Thank you,
Melike

**Reply by beat-buesser** [reply]
Hi @melikedem
I forgot to add the link to the method's paper: https://arxiv.org/pdf/1906.03849.pdf.
Algorithm 1+2 are implemented in `RobustnessVerificationTreeModelsCliqueMethod`.
Algorithm 3 in the paper's appendix is for a single decision tree.
Let us know if you would like to try to implement it for ART, it would be a great tool and we would be happy provide support!
